CONTRIBUTING.md

```diff
@@ -7,46 +7,82 @@ Want to contribute to Sitecore JavaScript Services? There are a few things you n
+```shell
+cd ./packages/sitecore-jss
+npm run lint
+```
 
-## Edit on GitHub
+Some linting errors may be fixed automatically using the `--fix` flag. You may do this from the root, however you will need `--` twice before the flag in order to pass the argument to the inner command
 
-Every route on the documentation site (https://jss.sitecore.com) has a link at the bottom of the page titled `Edit this on GitHub`. Clicking this link will use the GitHub editing features to allow you to make changes to the page content.
+```shell
+npm run lint-packages -- -- --fix 
+```
 
-When your changes are complete, use the `Create a new branch for this comment and start a pull request` option. This will create a pull request to the `master` branch. Your changes will be reviewed and merged if appropriate.
+In addition to linting rules, there are a few coding guidelines worth mentioning here that will cause less friction when trying to get a PR merged.
+
+- TypeScript: make sure everything has the appropriate type.
+
+- JSDoc: new functions, interfaces, classes, etc. need to have a JSDoc style comment above their declaration. Example: https://jsdoc.app/howto-es2015-classes.html
+
+## Run unit tests
 
-## Edit in repo
+To keep everything running smoothly, please include unit tests when applicable.
+To run all tests, from the root of the monorepo:
 
-If you need to make code changes to the documentation site, follow these steps:
+```shell
+npm run test-packages
+```
+
+In the root of a package:
+
+```shell
+npm run test
+```
+
+# Troubleshooting
+
+Problem: Build of package or sample fails with the following error: `cannot find module ...`
+
+Solution: If dependencies in a package change, you may need to reset the repo to have the latest changes. This is often needed after pulling changes that include a dependency change.
+From the root of the monorepo:
+
+```shell
+npm run reset
+```
+
+# Branching overview
 
-- Fork the JSS GitHub repo: `https://github.com/Sitecore/jss`
-- Clone the forked repo to your local machine.
-- From a terminal in the repo root: `cd docs`
-- Create a feature branch from `master` for your changes. e.g. `git checkout -b my-feature-branch`
-  > Note: although `dev` is the default branch for the JSS repo, documentation changes that pertain to the current release of JSS should be branched from `master`. The `master` branch contains the published / active documentation content. Changes related to the next release should be added to the `dev` branch.
-- `npm install`
-- `jss start` (to preview your changes locally)
-- Make your changes
-- Commit, push to your remote fork of the JSS repo, then open a pull request (PR) to the `master` branch of the JSS repo.
+* `master `- latest released version
+* `dev `- latest changes for the next release
+* `release `branches - created in order to make changes/updates to past major version releases
```

Comment:
These are also used for sample sources (e.g. `jss create my-app nextjs --branch release/16.0.0`). Not sure if we need to include that necessarily, but maybe we can simplify to something like:
```suggestion
* `release `branches - all major released versions
```

Reply:
Good call, changed.